### PR TITLE
Implement checks for test/semantics/call02.f90

### PR DESCRIPTION
**Pull request by klausler**
More semantic checking for calls.  This PR also implements enough analysis of coindexed references to catch an error condition in the test.

### CODE REVIEW

File: lib/semantics/expression.cc [hunk 1]
```diff
@@ -1515,18 +1564,31 @@ std::optional<ActualArgument> ExpressionAnalyzer::AnalyzeActualArgument(
...
-  if (const Symbol * assumedTypeDummy{AssumedTypeDummy(var)}) {
-    return ActualArgument{ActualArgument::AssumedType{*assumedTypeDummy}};
-  } else if (MaybeExpr argExpr{Analyze(var)}) {
-    return ActualArgument{std::move(*argExpr)};
+    Expr<SomeType> x{Fold(GetFoldingContext(), std::move(*argExpr))};
+    if (const auto *proc{std::get_if<ProcedureDesignator>(&x.u)}) {
+      if (!std::holds_alternative<SpecificIntrinsic>(proc->u) &&
+          proc->IsElemental()) {  // C1533
+        Say(expr.source,
+            "Non-intrinsic ELEMENTAL procedure cannot be passed as argument."_err_en_US);
```
Comment (tskeith) [inline lib/semantics/expression.cc]: We haven't been putting periods at the end of error messages.

File: lib/semantics/expression.cc [hunk 2]
```diff
@@ -1515,18 +1564,31 @@ std::optional<ActualArgument> ExpressionAnalyzer::AnalyzeActualArgument(
...
+      }
+    }
+    if (auto coarrayRef{ExtractCoarrayRef(x)}) {
+      const Symbol &coarray{coarrayRef->GetLastSymbol()};
+      if (const semantics::DeclTypeSpec * type{coarray.GetType()}) {
+        if (const semantics::DerivedTypeSpec * derived{type->AsDerived()}) {
+          if (auto ptr{semantics::FindPointerUltimateComponent(*derived)}) {
+            if (auto *msg{Say(expr.source,
+                    "Coindexed object '%s' with POINTER ultimate component '%s' cannot be passed as argument."_err_en_US,
```
Comment (tskeith) [inline lib/semantics/expression.cc]: Same here.

File: test/semantics/call02.f90 [hunk 3]
```diff
@@ -63,22 +63,6 @@ elemental real function elem04(x)
...
-    subroutine altreturn(*)
-    end subroutine
-  end interface
- contains
-  subroutine test
-1   continue
-   contains
-    subroutine internal
-      !ERROR: alternate return label must be in the inclusive scope
-      call altreturn(*1)
```
Comment (tskeith) [inline test/semantics/call02.f90]: Is this check still a TODO? Should the test be moved to a different call??.f90?

Comment (klausler) [reply test/semantics/call02.f90]: It was redundant; label validation catches the error, but then none of the other semantic checks are applied.

File: lib/semantics/expression.cc (hunk 1 above)
Comment (klausler) [reply lib/semantics/expression.cc]: Ok.

**Pull request merged**